lesson5/numbers_generator.py

```diff
@@ -0,0 +1,13 @@
+import random
+import os
+
+file_with_num = os.path.abspath('.')
+tuple_with_num = ()
+for num in range(0, 100):
+    left_operand = str(random.randint(1, 100))
+    right_operand = str(random.randint(1, 100))
+    operator = str(random.randint(1, 3))
+    tuple_with_num = (left_operand, right_operand, operator)
+    my_str = ' '.join(tuple_with_num)
+    with open('file_with_num', 'a') as file:
+        file.write(my_str + '\n')
```

Comment:
it works and is now better, but where is the list of tuples? Let's optimize. 

You perform open() every loop iteration which is not good practice. 

Just form a list of tuples, and then open file for writing ar appending. 

with open() as file:
     for data in my_list:
            write

So the algorithm will be next:

generate a list of tuples.
than open file for writing
loop list_of_tuples and write elements if format you need